### PR TITLE
Add --archive option to the build command.

### LINKED ISSUE #1411: Simpler API for carthage build --no-skip-current
- `carthage version`: 0.16.2
- `xcodebuild -version`: Xcode 7.3.1 Build version 7D1014
- Are you using `--no-build`? No
- Are you using `--no-use-binaries`? No
- Are you using `--use-submodules`? No
# Feature request:

I have to Google every single time I use `carthage build --no-skip-current` is there a simpler API for `carthage build --no-skip-current`? Something more memorable like:

`carthage build --archive`

So then I can do

`carthage archive`

Or maybe

`carthage archive --building`, so I can skip the build command?

Thanks for your help.

### PR DESCRIPTION
**Pull request by sunshinejr**
Fixes #1411. 

To be honest I just finished the works by @3lvis so credits to him for starting that one 😉

### CODE REVIEW

**Comment by blender** [comment]
So this is an alias for `--no-skip-current`?

**Comment by sunshinejr** [comment]
@blender I've updated the PR, should be ready for a review!

File: Source/carthage/Build.swift [hunk 1]
```diff
@@ -45,6 +59,7 @@ public struct BuildCommand: CommandProtocol {
 				<*> mode <| Option(key: "verbose", defaultValue: false, usage: "print xcodebuild output inline")
 				<*> mode <| Option(key: "project-directory", defaultValue: FileManager.default.currentDirectoryPath, usage: "the directory containing the Carthage project")
 				<*> mode <| Option(key: "log-path", defaultValue: nil, usage: "path to the xcode build output. A temporary file is used by default")
+				<*> mode <| Option(key: "archive", defaultValue: false, usage: "don't skip building the Carthage project (in addition to its dependencies) and archive")
```
Comment (mdiep) [inline Source/carthage/Build.swift]: I think this description could use a little work.

Maybe:

```
"archive built frameworks from the current project (implies --no-skip-current)"
```

?

**Comment by mdiep** [comment]
This is great! Thanks for the PR!

I think we should tweak the usage description, but otherwise this should be ready to g.

File: Source/carthage/Build.swift (hunk 1 above)
Comment (sunshinejr) [reply Source/carthage/Build.swift]: This is a good one! Updating it as we speak.

**Comment by sunshinejr** [comment]
Thanks, @mdiep! Updated the description, let me know if there is anything else that can be improved here.

**Comment by mdiep** [comment]
Thanks @sunshinejr !

**Comment by sunshinejr** [comment]
@3lvis @mdiep Thanks guys! Was fun to help, even just a little bit :)

**Pull request merged**